### PR TITLE
[Storage] Allow setting http headers for default http client 

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
Exposes a config in storage config section to allow setting http headers to the default http client. This is because some of the underlying storage systems (e.g. s3 stow) doesn't allow setting some settings (e.g. ACL) and the only way to support that is through setting HTTP HEADERS that AWS S3 respects.

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [X] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/261

### CODE REVIEW

File: storage/config.go [hunk 1]
```diff
@@ -52,8 +52,9 @@ type Config struct {
 	// inputs is accelerated. The size of the cache is large so understand how to configure the cache.
 	// TODO provide some default config choices
 	// If this section is skipped, Caching is disabled
-	Cache  CachingConfig `json:"cache"`
-	Limits LimitsConfig  `json:"limits" pflag:",Sets limits for stores."`
+	Cache                    CachingConfig       `json:"cache"`
+	Limits                   LimitsConfig        `json:"limits" pflag:",Sets limits for stores."`
+	DefaultHTTPClientHeaders map[string][]string `json:"defaultHttpClientHeaders" pflag:"-,Sets http headers to set on the default http client."`
```
Comment (kumare3) [inline storage/config.go:57]: do you want to create a separate section called client-defaults?
headers: ....
etc

File: storage/config.go [hunk 2]
```diff
@@ -52,8 +52,15 @@ type Config struct {
 	// inputs is accelerated. The size of the cache is large so understand how to configure the cache.
 	// TODO provide some default config choices
 	// If this section is skipped, Caching is disabled
-	Cache  CachingConfig `json:"cache"`
-	Limits LimitsConfig  `json:"limits" pflag:",Sets limits for stores."`
+	Cache             CachingConfig     `json:"cache"`
+	Limits            LimitsConfig      `json:"limits" pflag:",Sets limits for stores."`
+	DefaultHTTPClient *HTTPClientConfig `json:"defaultHttpClient" pflag:",Sets the default http client config."`
```
Comment (wild-endeavor) [inline storage/config.go:57]: this makes it sound like a client, not a config for a client.

**Comment by wild-endeavor** [comment]
up to you on the name, let me know when you want a +1

File: storage/config.go (hunk 2 above)
Comment (EngHabu) [reply storage/config.go:57]: Similar to Cache?
